File: Stack/stack.py
class Stack(object):

    def __init__(self,size):
        self.index = -1
        self.stack = [None] * size
    
    def push(self, value):
        if self.index != len(self.stack) -1:
            self.index = self.index + 1
            self.stack[self.index] = value

    def pop(self):
        if not self.empty():
            value = self.stack[self.index]
            self.stack[self.index] = None
            self.index = self.index - 1
            return value

    def peek(self):
        if not self.empty():
            return self.stack[self.index]

    def empty(self):
        return self.index == -1

File: Stack/stack.js
/**
 * Implementation of stack with an array
 */

class Stack {
  constructor() {
    this.stack = [];
  }
  
  // Insert an element on top of the stack
  push(element) {
    this.stack.push(element);
  }
  
  //Removes the element at the top of the stack and returns that same element
  pop() {
    if (this.isEmpty()) 
    return 'Stack is empty!';
    return this.stack.pop();
  }
  
  // Returns the element that is on top of the stack
  peek() {
    if (this.isEmpty()) 
    return 'Stack is empty';
    return this.stack[this.stack.length - 1];
  }
  
  // helper method
  isEmpty() {
    return !this.stack.length;
  }
}